File: palindrome-number/palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x<0){
            return false;
        }
        vector<int> v;
        while(x){
            v.push_back(x%10);
            x=x/10;
        }
        int i=0,j=v.size()-1;
        while(i<j){
            if(v[i++]!=v[j--]){
                return false;
            }
        }
        return true;
    }
};

File: palindrome-number/palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if(x < 0){
            return false;
        }
        int original = x;
        int reverse = 0;
        while(x > 0){
            int temp = x%10;
            x = x/10;
            reverse = reverse*10 + temp;
        }
        return original == reverse;
    }
}